File: sdks/ts/src/sdk/models/shared/legalholdpolicy.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UserMini } from "./usermini";



// LegalHoldPolicyAssignmentCounts
/** 
 * Counts of assignments within this a legal hold policy by item type
**/
export class LegalHoldPolicyAssignmentCounts extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=file" })
  file?: number;

  @SpeakeasyMetadata({ data: "json, name=file_version" })
  fileVersion?: number;

  @SpeakeasyMetadata({ data: "json, name=folder" })
  folder?: number;

  @SpeakeasyMetadata({ data: "json, name=user" })
  user?: number;
}

export enum LegalHoldPolicyStatusEnum {
    Active = "active",
    Applying = "applying",
    Releasing = "releasing",
    Released = "released"
}

export enum LegalHoldPolicyTypeEnum {
    LegalHoldPolicy = "legal_hold_policy"
}


// LegalHoldPolicy
/** 
 * A mini legal hold policy
**/
export class LegalHoldPolicy extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=assignment_counts" })
  assignmentCounts?: LegalHoldPolicyAssignmentCounts;

  @SpeakeasyMetadata({ data: "json, name=created_at" })
  createdAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=created_by" })
  createdBy?: UserMini;

  @SpeakeasyMetadata({ data: "json, name=deleted_at" })
  deletedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=filter_ended_at" })
  filterEndedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=filter_started_at" })
  filterStartedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=modified_at" })
  modifiedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=policy_name" })
  policyName?: string;

  @SpeakeasyMetadata({ data: "json, name=release_notes" })
  releaseNotes?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: LegalHoldPolicyStatusEnum;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: LegalHoldPolicyTypeEnum;
}

File: sdks/python/src/boxsdk/models/shared/legalholdpolicy.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_mini as shared_user_mini


@dataclass_json
@dataclasses.dataclass
class LegalHoldPolicyAssignmentCounts:
    r"""LegalHoldPolicyAssignmentCounts
    Counts of assignments within this a legal hold policy by item type
    """
    
    file: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('file') }})
    file_version: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('file_version') }})
    folder: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('folder') }})
    user: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('user') }})
    
class LegalHoldPolicyStatusEnum(str, Enum):
    ACTIVE = "active"
    APPLYING = "applying"
    RELEASING = "releasing"
    RELEASED = "released"

class LegalHoldPolicyTypeEnum(str, Enum):
    LEGAL_HOLD_POLICY = "legal_hold_policy"


@dataclass_json
@dataclasses.dataclass
class LegalHoldPolicy:
    r"""LegalHoldPolicy
    A mini legal hold policy
    """
    
    assignment_counts: Optional[LegalHoldPolicyAssignmentCounts] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('assignment_counts') }})
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    created_by: Optional[shared_user_mini.UserMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_by') }})
    deleted_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deleted_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    filter_ended_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filter_ended_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    filter_started_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filter_started_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    modified_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modified_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    policy_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('policy_name') }})
    release_notes: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('release_notes') }})
    status: Optional[LegalHoldPolicyStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    type: Optional[LegalHoldPolicyTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
